woocommerce/woocommerce
# Add improved tracking to the marketplace

### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/woocommerce.com/issues/18294

The issue linked provides improvements to the interaction that are tracked in the marketplace page. The initial priority is to ensure that all the tracking that currently exists in the PHP marketplace, is still present in the React marketplace. This PR handles that and adds some improvement to that tracking to better suit the new marketplace structure.



### How to test the changes in this Pull Request:



1. Checkout this branch and build the project
2. Visit any WC Admin page such as `/wp-admin/admin.php?page=wc-admin` ([wp-env link](http://localhost:8888/wp-admin/admin.php?page=wc-admin))
3. Open the browser console, the paste and run the following: `localStorage.setItem( 'debug', 'wc-admin:*' );` to allow tracks events from React to show in the console.
4. Visit the marketplace pages and see the various events are recording correctly to the console. They should make sense based on the tables below. Note: the legacy events are mapped to the logic of the events listed [here](https://github.com/woocommerce/woocommerce/blob/9a0e91477eadc4b4a6027197edb4a5ce1d48548c/plugins/woocommerce/includes/tracks/events/class-wc-extensions-tracking.php#L31-L48).
5. Particularly, check for doubling of event triggering that may skew data.
6. Test around the issue

**Legacy events structure**
| Event | Name | Properties |
|--------|--------|--------|
| Discover tab | `extensions_view` | `section: '_featured'` `version: '2'`|
| Browse (extensions) tab | `extensions_view` |  `section: '_all'` `version: '2'` |
| Browse tab with category selected | `extensions_view` |  `section: category name` `version: '2'` |
| Themes tab | `extensions_view` |  `section: 'themes'` `version: '2'` |
| Themes tab with category selected | no event since this isn't currently supported | N/A |
| Search results | `extensions_view_search` |  `section: 'search'` `search_term: search query entered` `version: "2"` |
| Search results > View All button | no event since this isn't currently supported | N/A |
| Search results > View All button with category | no event since this isn't currently supported | N/A |
| Subscriptions tab | `subscriptions_view` |  `section: 'helper'` `version: '2'` |

**New events structure**
Note, all events have the same name – `marketplace_v2_view` so the name column isn't shown.
| Event | Properties |
|--------|--------|
| Discover tab | `view: 'discover'` |
| Browse (extensions) tab | `view: 'extensions'` |
| Browse tab with category selected | `view: extensions` `category: category name` |
| Themes tab | `view: 'themes'` |
| Themes tab with category selected | `view: 'themes'` `category: category name` |
| Search results | `view: 'search'` `searchTerm: search query entered` |
| Search results > View All button | `view: 'search'` `searchTerm: search query entered` `productType: 'extension' or 'theme'` |
| Search results > View All button with category | `view: 'search'` `searchTerm: search query entered` `productType: 'extension' or 'theme'` `category: category name` |
| Subscriptions tab | `view: 'my-subscriptions'` |



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Ensure new marketplace works with the existing tracking events.

</details>
